e2e/test/provisioning/ProvisioningServiceClientE2ETests.cs

```diff
@@ -149,98 +149,132 @@ public async Task ProvisioningServiceClient_GetIndividualEnrollmentAttestation(A
+                {
+                    attestationMechanism.Type.Should().Be(AttestationMechanismType.SymmetricKey);
+
+                    var symmetricKeyAttestation = (SymmetricKeyAttestation)attestationMechanism.GetAttestation();
+                    symmetricKeyAttestation.PrimaryKey.Should().Be(((SymmetricKeyAttestation)individualEnrollment.Attestation).PrimaryKey);
+                    symmetricKeyAttestation.SecondaryKey.Should().Be(((SymmetricKeyAttestation)individualEnrollment.Attestation).SecondaryKey);
+                }
+                else if (attestationType == AttestationMechanismType.X509)
+                {
+                    attestationMechanism.Type.Should().Be(AttestationMechanismType.X509);
 
-                var x509Attestation = (X509Attestation)attestationMechanism.GetAttestation();
-                x509Attestation.GetPrimaryX509CertificateInfo().SHA1Thumbprint.Should().Be(((X509Attestation)individualEnrollment.Attestation).GetPrimaryX509CertificateInfo().SHA1Thumbprint);
-                x509Attestation.GetSecondaryX509CertificateInfo().SHA1Thumbprint.Should().Be(((X509Attestation)individualEnrollment.Attestation).GetSecondaryX509CertificateInfo().SHA1Thumbprint);
+                    var x509Attestation = (X509Attestation)attestationMechanism.GetAttestation();
+                    x509Attestation.GetPrimaryX509CertificateInfo().SHA1Thumbprint.Should().Be(((X509Attestation)individualEnrollment.Attestation).GetPrimaryX509CertificateInfo().SHA1Thumbprint);
+                    x509Attestation.GetSecondaryX509CertificateInfo().SHA1Thumbprint.Should().Be(((X509Attestation)individualEnrollment.Attestation).GetSecondaryX509CertificateInfo().SHA1Thumbprint);
+                }
+                else
+                {
+                    attestationMechanism.Type.Should().Be(AttestationMechanismType.Tpm);
+
+                    var tpmAttestation = (TpmAttestation)attestationMechanism.GetAttestation();
+                    tpmAttestation.EndorsementKey.Should().Be(((TpmAttestation)individualEnrollment.Attestation).EndorsementKey);
+                    tpmAttestation.StorageRootKey.Should().Be(((TpmAttestation)individualEnrollment.Attestation).StorageRootKey);
+                }
             }
-            else
+            finally
             {
-                attestationMechanism.Type.Should().Be(AttestationMechanismType.Tpm);
-
-                var tpmAttestation = (TpmAttestation)attestationMechanism.GetAttestation();
-                tpmAttestation.EndorsementKey.Should().Be(((TpmAttestation)individualEnrollment.Attestation).EndorsementKey);
-                tpmAttestation.StorageRootKey.Should().Be(((TpmAttestation)individualEnrollment.Attestation).StorageRootKey);
+                if (individualEnrollment != null)
+                {
+                    await DeleteCreatedEnrollmentAsync(EnrollmentType.Individual, individualEnrollment.RegistrationId, null);
```

Comment:
we'll also want to dispose any certificates associated with the x509 attestation mechanism

Reply:
same for the group enrollment as well

Reply:
The provisioning service E2E tests don't seem to be creating an x509 certificate based enrollments, so you can either add those or add a comment calling out that we're not testing that in here.
Iirc we cover x509 certificate based scenarios pretty extensively in the provisioning device client E2E tests so I'd be fine with us simply adding a comment here calling out that we're not testing x509 here.

Reply:
Added comment and removed unused x509 code